internal/credential/repository_store.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/pagination"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// SubtypeStoreService defines the interface expected
+// to gather information about credential stores.
+type SubtypeStoreService interface {
+	ListDeletedStoreIds(context.Context, time.Time, ...Option) ([]string, error)
+}
+
+// StoreRepository coordinates calls across different subtype services
+// to gather information about all credential stores.
+type StoreRepository struct {
+	services []SubtypeStoreService
+	writer   db.Writer
+}
```

Comment:
I called this a repository now, since it accesses the DB. I had a look at the DDD definitions of Service and Repository and this feels more like a repository now. Thoughts appreciated.